lib/commonFunctions.js

```diff
@@ -19,6 +19,42 @@ var sundial = require('sundial');
 
 var annotate = require('./eventAnnotations');
 
+/**
+ * Computes the number of milliseconds after midnight on the date specified.
+ *
+ * @param dateTime DateTime object to figure out millis from
+ * @returns {number} number of millis in current day
+ */
+exports.computeMillisInCurrentDay = function(e){
+  var msFromMidnight = sundial.getMsFromMidnight(e.time, e.timezoneOffset);
```

Comment:
Just thought of another thing to point you to that may help explain...

Well first, remember that the "equation" for timestamps is this (from the BtUTC doc):
```
deviceTime = time + timezoneOffset + conversionOffset
```

Notice that `clockDriftOffset` is not part of it. We track it as a remainder from the rounding we do with `timezoneOffset`s, but that's it. And if you compare [TZOUtil line 67](https://github.com/tidepool-org/chrome-uploader/blob/master/lib/TimezoneOffsetUtil.js#L67) with [TZOUtil line 70](https://github.com/tidepool-org/chrome-uploader/blob/master/lib/TimezoneOffsetUtil.js#L70) you'll see that we don't do the same remaindering process with `conversionOffset`, so we are actually essentially *including* the `clockDriftOffset` when a `conversionOffset` is involved.

The resulting messy logic in `computeMillisInCurrentDay` is because of this difference: when `conversionOffset` is nonzero, `clockDriftOffset` is included in the calculation of `time`, which makes matching up against schedules harder.

You might ask: does this suggest that we took the wrong approach with `conversionOffset`? (I am asking myself this 😛) But I think from a data accuracy standpoint we can't do any better. `conversionOffset`s result from large changes in display time due to incorrect pump set up. The idea of separating out part of that change as "clock drift" (by doing the same rounding and remaindering process we do with `timezoneOffset`) feels really artificial and arbitrary.

/end more-than-you-wanted-to-know